### DISCUSSION TITLE
Is there an option to show the hamburger left menu by default?

### DISCUSSION THREAD

**Discussion by jrichardsz** [squidfunk/mkdocs-material#3210] [Q&A] [answered]
I have several sub folders and the default menu is not responsive when screen is big:

![image](https://user-images.githubusercontent.com/3322836/141387045-9f0e0480-793c-4ca3-aef5-f11195ee57ae.png)

But if I reduce mi screen, the hamburger menu appear 

![image](https://user-images.githubusercontent.com/3322836/141387133-fb004455-2c0a-4c31-a838-fb92bbec79eb.png)

And an awesome infinite menu appear:
![image](https://user-images.githubusercontent.com/3322836/141387118-fcc54b76-5c77-4e0f-8ffd-cca1c19ce46c.png)

So, my question is: How can I set the hamburger menu as default instead the plain left menu.

Thanks

**Comment by squidfunk** [comment, marked as answer]
There's no option. You would need to patch the theme and adjust the media queries. For starters:

https://github.com/squidfunk/mkdocs-material/blob/38e001313ab79b2d65f7755491022243867572c8/src/assets/stylesheets/_config.scss#L28-L42

https://github.com/squidfunk/mkdocs-material/blob/38e001313ab79b2d65f7755491022243867572c8/src/assets/javascripts/bundle.ts#L94-L95

Also, search the source for `1220px` – there might be 1-2 further references. You need to increase that value, so the logic kicks in later (or never, if you wish).

**Reply by saket** [reply]
Can these berakpoints be made configurable by any chances?

I'm trying to make my website look better on foldable devices by not hiding the left sidebar especially when TOC is already integrated with the navigation bar.

**Reply by saket** [reply]
Alternatively, can these be overridden using `custom_dir`? I tried adding a `overrides/assets/stylesheets/_config.scss` but it doesn't seem to get picked up.

**Reply by squidfunk** [reply]
You need to go through the [theme development process](https://squidfunk.github.io/mkdocs-material/customization/#theme-development), apply your changes and rebuild the theme. Sadly, we cannot make those breakpoints configurable via `mkdocs.yml`, as I've already explained multiple times, for example in https://github.com/squidfunk/mkdocs-material/discussions/3210#discussioncomment-2342214

**Reply by saket** [reply]
Yep you did and I conveniently overlooked that message. Sorry!

**Comment by toomanybrians** [comment]
I'm trying to do a similar thing, where I want *not* flip to the hamburger menu too soon. Currently when the browser width goes to less than 1220px, it flips to the hamburger, but I want to shrink the site down to like 1000px or so. (My personal web browsing width is less than 1220 but I want the desktop interface.) I have searched through the discussions and been looking through the CSS and stuff. Is there any way I can make this behavior happen with CSS overrides alone, or do I have to actually change the theme source and recompile it?

Thanks!
Brian

**Reply by squidfunk** [reply]
I'm afraid there's no other way than patching the theme by forking and recompiling it.

**Reply by StefanoCecere** [reply]
arg. this is a bit unfortunate... 1220 is a bit too much, since i use navigation horinzontal menu e lot.
i was looking how to configure it, but i'm not going to recompile the theme...

**Reply by squidfunk** [reply]
Well, it's not possible without recompilation. If, one day, [`@custom-media`](https://drafts.csswg.org/mediaqueries-5/#custom-mq) is supported by >90% of browsers, we can re-evaluate, but I don't think that'll happen any time soon. Until then, it's just technically impossible to provide this as a configuration option.

**Reply by MaximilianKohler** [reply]
@squidfunk would it be possible for you to just decrease the default? I agree that 1220 is far too wide/early to be removing the bars. Maybe half that would be good... The way this works is perfect: [https://hugo-book-demo.netlify.app/](https://hugo-book-demo.netlify.app/docs/shortcodes/hints/) - and after it closes it puts shortcuts at the top. Only thing I'd change with that is the shortcuts scroll with the page; they shouldn't. 

P.S. they're open source too and their search works quite well.

**Reply by squidfunk** [reply]
@MaximilianKohler you're proposing a pretty fundamental change. We can't simply change this behavior, given how many installations are already out there. FWIW, 1220 is not randomly chosen – the whole site is 1200px wide and the 20px are necessary for padding the sidebars so they always keep enough space to the viewport. On desktop, the content area's width is usually not longer than 75 characters, which is [the ideal width for reading](https://www.google.com/search?q=characters+per+line+ideal), and can host exactly 80 characters in code blocks at the same time, which is also a more or less de-facto standard. When we would keep the sidebars on smaller screens, the content area would be too narrow to be useful and the whole thing would look very weird. See your example:

![screenshot-hugo-book-demo-netlify-app-docs-shortcodes-hints-1676027269371](https://user-images.githubusercontent.com/932156/218078380-e322d7a1-e631-4d52-9658-77b7cfdfb3d5.png)

This just doesn't look right. Yes, this is _my_ opinion, but that's what this project is: an opinionated template. As mentioned before, [this behavior can be changed by customization](https://github.com/squidfunk/mkdocs-material/discussions/3210#discussioncomment-1630367), so you're invited to fork the theme if you don't like it.

**Reply by MaximilianKohler** [reply]
Ok. Personally I don't think that screenshot looks weird, but there are other ways this could be solved:
* You could make the right-side TOC hide first. It's easy to add a manual TOC at the top of a page. 
* You could even make the right-side TOC switch to the top instead of completely hide. 
* You could add a TOC icon to the top right once the TOC hides - like that Hugo Book theme does. 
* An option to let us choose whether to hide the TOC first or the left side.

**Reply by squidfunk** [reply]
Jup, this can all be done with [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme). We won't add it to `master` for all the reasons stated.

**Reply by MaximilianKohler** [reply]
Thanks Squid! I checked that link and I can't find instructions to make the types of changes I listed above. I guess it's only for advanced users to figure out for themselves?

**Reply by squidfunk** [reply]
The customization guide explains _how_ to extend Material for MkDocs to add functionality that is not provided by the theme itself. The guide is arranged in increasing levels of involvement, starting with just changing a few styles or adding extra functionality via third-party JavaScript libraries, up to changing theme fundamentals and recompiling it. You might achieve your desired outcome with a little knowledge of CSS and JavaScript, and by following the theme development process.

**Reply by andrzejdoro** [reply]
I would like to have the same behavior as described by @MaximilianKohler. In my use case, I keep the right TOC hidden by default.
I use the following header in the .md file.
```
---
hide:
  - toc
---
```

I understand the concept of having about 80 characters visible. However, when the right panel is hidden then there is enough space to reduce the page below 1220px while keeping the left navigation visible.

Additionally, using the popular 1920px resolution, I personally find it inconvenient to keep the browser at about 2/3 the width of the monitor to still have navigation at hand. It would be really helpful if the ability to change the default width behavior was relatively simple and documented.

**Reply by squidfunk** [reply]
> It would be really helpful if the ability to change the default width behavior was relatively simple and documented.

I understand. [Please read the entire conversation](https://github.com/squidfunk/mkdocs-material/discussions/3210#discussioncomment-2342214) why this is currently not "relatively simple and documented".

**Reply by saket** [reply]
I like the idea of hiding TOC and reclaiming that space for the navigation sidebar.

**Reply by ghudgins** [reply]
1200 is just so so wide. we watch users try our product for the first time and very frequently observe people on desktop monitors putting docs on one half and a terminal or IDE in the other and they don't get the navigation. this theme is so close to being great! (FWIW I do like that it's always visible - saves users a click to navigate around technical documentation)

if you wanted to see what I mean.... https://flox.dev/docs/

**Reply by squidfunk** [reply]
We're actively working on making SASS compilation a part of the build process of documentation, which means users will be able to change those values. It's also part of a larger effort, as mentioned in [our latest blog post](https://squidfunk.github.io/mkdocs-material/blog/2024/08/19/how-were-transforming-material-for-mkdocs/).

**Reply by pneyrinck** [reply]
Thank you for the hard work solving this. My project was trying to use Mkdocs within an integrated software application where users need docs to fit in next to other sub windows. The left column with integrated TOC is a wonderful feature. But, anything less than 1220px hides that lovely interface.

**Comment by jrichardsz** [comment]
I tried replacing specific sizes based on the suggested of squidfunk in the bundle unlucky :(
